http/src/main/java/io/crate/http/netty/CrateNettyHttpServerTransport.java

```diff
@@ -76,11 +75,41 @@ public ChannelPipelineFactory configureServerChannelPipelineFactory() {
     }
 
     public void addBefore(ChannelPipelineItem item) {
-        addBeforeList.add(item);
+        synchronized (addBeforeList) {
+            addBeforeSorted(addBeforeList, item);
+        }
+    }
+
+    List<ChannelPipelineItem> addBefore() {
+        return addBeforeList;
     }
 
-    public void addAfter(ChannelPipelineItem item) {
-        addAfterList.add(item);
+    private void addBeforeSorted(List<ChannelPipelineItem> addBeforeList, ChannelPipelineItem newItem) {
+        addBeforeList.add(newItem);
+
+        ArrayList<ChannelPipelineItem> copy = new ArrayList<>(addBeforeList.size());
+        copy.addAll(addBeforeList);
+
+        for (ChannelPipelineItem item : copy) {
+            addBeforeList.remove(item);
+
+            Optional<ChannelPipelineItem> prev = addBeforeList.stream()
+                .filter(o -> o.name.equals(item.base))
+                .findFirst();
```

Comment:
I don't think it is possible to implement using a sort comparator, because I need to check against the whole array,
e.g. an item (format: `base->name`) `a->b` is not comparable to `d->e`, because they don't have a common denominator to compare